### ISSUE TITLE
`reportAttributeAccessIssue` reports error when assigning to unexistent class variable

### ISSUE THREAD

**Issue by some0necoding** [microsoft/pyright#9726]
**Describe the bug**
`reportAttributeAccessIssue` complains about "foo" being unknown even if this is a legal assignment in python, useful for example to have static instances of a class.

**Code or Screenshots**
```python
class Foo:
    def __init__(self) -> None:
        pass

Foo.foo = "foo" # cannot assign to attribute "foo" for class "type[Foo]" Attribute "foo" is unknown
Foo.static_instance = Foo() # cannot assign to attribute "static_instance" for class "type[Foo]" Attribute "static_instance" is unknown
```

**VS Code extension or command-line**
Running as language server in neovim.
Python version: 3.13.1
Pyright version: 1.1.391

**Comment by erictraut** [comment]
Pyright is working correctly here, so this isn't a bug.

Static type checkers like pyright or mypy will prevent you from accessing attributes on a class if those attributes are not declared within the class implementation. This is by design, as it's intended to help prevent unintended accesses.

You can declare the attribute in the class body as follows.

```python
from typing import ClassVar

class Foo:
    foo: str
    static_instance: "ClassVar[Foo]"

    def __init__(self) -> None:
        pass

Foo.foo = "foo"
Foo.static_instance = Foo()
```
